### PR TITLE
Return `404` in the API if the requested webhooks were not found

### PR DESCRIPTION
**Pull request by sonjek**
Should resolve first point of the issue https://github.com/go-gitea/gitea/issues/24574

### CODE REVIEW

File: routers/api/v1/admin/hooks.go [hunk 1]
```diff
@@ -74,7 +74,11 @@ func GetHook(ctx *context.APIContext) {
 	hookID := ctx.ParamsInt64(":id")
 	hook, err := webhook.GetSystemOrDefaultWebhook(ctx, hookID)
 	if err != nil {
-		ctx.Error(http.StatusInternalServerError, "GetSystemOrDefaultWebhook", err)
+		if webhook.IsErrWebhookNotExist(err) {
```
Comment (delvh) [inline routers/api/v1/admin/hooks.go]:
```suggestion
		if errors.Is(err, util.ErrNotExist) {
```
additionally needs the corresponding imports

Comment (sonjek) [reply routers/api/v1/admin/hooks.go]: fixed in https://github.com/go-gitea/gitea/pull/24823/commits/954abcd8c577976065bbab4cf3b6a23fcb740574

**Review by delvh — APPROVED** [review]

**Comment by sonjek** [comment]
@delvh I added a test. How can I test that it works?

**Comment by delvh** [comment]
We wait for the CI icons to turn green (except for `giteabot/lgtm`, that will only turn green with the second approval).
That's the easiest approach.

**Comment by delvh** [comment]
Otherwise, you can also try a local `make test-backend` or something like that.

**Review by techknowlogick — APPROVED** [review]
Thanks 🙏

**Comment by xanderxfz** [comment]
Hi, @lunny @delvh, can we backport this bugfix to 1.19.X?

**Comment by lunny** [comment]
@GiteaBot Please send backport to v1.19

**Comment by yardenshoham** [comment]
It'll do so on the next merge to main

**Pull request merged**